[ONNX] Generate diagnostic source files during build

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86949

Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

BowenBao committed:
[ONNX] Generate diagnostic source files during build

[ghstack-poisoned]

BowenBao committed:
Add onnx diagnostics codegen to build on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao committed:
fix mypy on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao committed:
Update bazel on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao committed:
fix bzl file path on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao committed:
bazel attempt on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao committed:
fix bazel path on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao committed:
add space for bzl build on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao commented:
> pull / linux-bionic-cuda11.6-py3.10-gcc7-bazel-test / build-and-test (pull_request) Failing after 8m
[Details](https://github.com/pytorch/pytorch/actions/runs/3252969977/jobs/5339801135)

Could use some help on this @kit1980. I'm updating `generate_code` to also generate some code for onnx exporter, could you let me know what I'm missing for bazel build?

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


- [ ] Fix bazel build
- [ ] Fix cmake for unnecessary regenerate/rebuild

[ghstack-poisoned]

BowenBao committed:
fix rebuild issue on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [ ] Fix cmake for unnecessary regenerate/rebuild

[ghstack-poisoned]

BowenBao committed:
remove unused file on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

[ghstack-poisoned]

BowenBao commented:
@kit1980 I figured out the bazel build. The PR is ready, please take a look.

cc @malfet

malfet reviewed (approved):
It would be nice to add a better description explaining what those diagnostic files are supposed to solve. Also, as it makes pretty non-trivial change to the build rules, I propose we go with a co-development workflow (i.e. I'm going to import it into internal build system and check if any fixes are needed

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40638423).

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

BowenBao commented:
@malfet friendly ping, any updates on progress?

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

BowenBao commented:
@malfet pop this PR again, are we blocked on anything?

BowenBao commented:
@pytorchbot rebase

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

kit1980 commented:
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40638423).

BowenBao commented:
Hi @kit1980 , do you have an estimate when this gets merged?

kit1980 commented:
@pytorchbot rebase

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

kit1980 commented:
@BowenBao `torch/csrc/onnx/diagnostics/diagnostics.h` still includes removed `torch/csrc/onnx/diagnostics/generated/rules.h`?
 This causes a failure internally, not sure why it wasn't caught by GitHub CI.

kit1980 commented:
@pytorchbot rebase

BowenBao committed:
Update on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

kit1980 commented:
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40638423).

BowenBao committed:
Add arguments for rule path and template dir on "[ONNX] Generate diagnostic source files during build"


Add the generation of diagnostics rules source code into build.

- [x] Fix bazel build
- [x] Fix cmake for unnecessary regenerate/rebuild

Differential Revision: [D40638423](https://our.internmc.facebook.com/intern/diff/D40638423)

[ghstack-poisoned]

BowenBao commented:
Replaced by https://github.com/pytorch/pytorch/pull/94257

thiagocrepaldi reviewed (commented):
Should we close this PR in favor of https://github.com/pytorch/pytorch/pull/94257
